docker/Dockerfile

```diff
@@ -145,12 +145,12 @@ RUN export DEBIAN_FRONTEND=noninteractive && \
 # - Second set of packages are recommended and/or required to build
 #   documentation or run tests.
 # - Third set of packages are optional but required for
-#   pygmsh/meshio/DOLFINX mesh pipeline.
+#   gmsh/meshio/DOLFINX mesh pipeline.
 RUN pip3 install --no-cache-dir mpi4py numba && \
     pip3 install --no-cache-dir cffi cppimport flake8 pytest pytest-xdist sphinx sphinx_rtd_theme && \
     export HDF5_MPI="ON" && \
     export CC=mpicc && \
-    pip3 install --no-cache-dir --no-binary=h5py h5py meshio pygmsh
+    pip3 install --no-cache-dir --no-binary=h5py h5py meshio
```

Comment:
Do we still need `meshio`?

Reply:
It is not a strict dependency any longer. However, to interface with msh-files, we would require it. Maybe we should create a separate PR with a msh->dolfin demo.

Reply:
Not sure I see the point. I would remove it if it isn't a test dependency. The few dependencies, the fewer things can can break a build.

Reply:
We need to have an interface to `msh` files at some point, as it is quite strict to require everyone to go through the `gmsh` interface.
In many cases, people might have msh files from other software.
The main problem is that meshio `msh`->`xdmf` is not directly compatible with dolfinx, see: https://fenicsproject.discourse.group/t/transitioning-from-mesh-xml-to-mesh-xdmf-from-dolfin-convert-to-meshio/412/79?u=dokken